src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultICollectionConverter.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Text.Json.Serialization.Policies;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    internal class JsonICollectionT<TSource> : ICollection<TSource>
+    {
+        private List<TSource> _list;
+        public JsonICollectionT(IList sourceList)
+        {
+            _list = new List<TSource>();
```

Comment:
I suppose we could just assign `_list` to `sourceList` for best performance here, and then do more work in each member here to cast to `TSource`

Reply:
`sourceList` would have to be `List<T>`. I'm adding a TODO so we can do a direct assignment.